### DISCUSSION TITLE
Column fixed prop

### DISCUSSION THREAD

**Discussion by whoslen-alt** [icflorescu/mantine-datatable#229] [Ideas]
I think there should be "fixed" or "fixed position" props in column options, which is pretty good UX for many columns table, wdy think?

**Comment by icflorescu** [comment]
Sticky columns are not exactly a breeze to implement, especially in conjunction with other generic Mantine DataTable features, because you don't really know "a priori" their widths, and when you position something with "sticky" you need to offset it relative to its nearest scrolling ancestor and containing block (see this [MDN docs page on positioning](https://developer.mozilla.org/en-US/docs/Web/CSS/position)). In case of sticky columns, we'd need to know each one's "left", and this would be quite difficult (if not impossible) to implement in a generic way that doesn't negatively impact other DataTable features.

In the Mantine DataTable component, the cell containing the selection checkbox for each row **is sticky**, but in that case it's not that difficult to implement because I know it's always the first cell, and therefore I could set its left to 0.

Have a look at the source code here to better understand what I mean: https://github.com/icflorescu/mantine-datatable/blob/main/package/DataTableRowSelectorCell.tsx.

That being said, I'm completely open to suggestions on how to implement this. Or better yet, a PR would be fantastic!
Just drop me a line on the email address in my GitHub profile if you think you could give it a try.

**Reply by nafetswirth** [reply]
Would it be possible to at least make the first column sticky similar to the last?

I think that's probably the most common use-case.

**Reply by icflorescu** [reply]
Funny you should say that; a week ago some people said pinning the **last** column would be the most common use-case 🤣
Hence my streak to implement the feature.

Doing the same for the 1st column would be a tad more difficult, since the 1st column is not always the 1st. There's also the checkbox column when using [record selection](https://icflorescu.github.io/mantine-datatable/examples/records-selection/), which **is** pinned when horizontal scrolling occur. At the moment, the selection column doesn't have a fixed width (so it can properly scale when using a non-standard Mantine size). Which means that there's no way to know where to "pin" the 1st data column that comes after it.

I'm not saying it can't be done, though.

A lot has been on my mind lately, including the V7 upgrade, which hasn't exactly been a breeze, do I don't have the time and 
energy to spend on this at the moment.

But like I said before, I'm not opposed to a PR implementing the feature.
And I'm also open to sponsorships from companies that are using Mantine DataTable and need this feature 😉
